frontend/src/components/unsubscribe/Unsubscribe.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect } from 'react'
+import { useLocation, useParams, Redirect } from 'react-router-dom'
+import querystring from 'querystring'
+
+import { ErrorBlock, PrimaryButton } from 'components/common'
+
+import styles from './Unsubscribe.module.scss'
+import appLogo from 'assets/img/brand/app-logo.svg'
+import landingHero from 'assets/img/unsubscribe/request-unsubscribe.png'
+
+import {
+  unsubscribeRequest,
+  isUserUnsubscribed,
+} from 'services/unsubscribe.service'
+
+const Unsubscribe = () => {
+  const { version } = useParams()
+  const location = useLocation()
+  const [errorMsg, setErrorMsg] = useState('')
+  const [isValid, setValid] = useState(true)
+  const [isUnsubscribed, setUnsubscribed] = useState(false)
+  const [campaignId, setCampaignId] = useState('')
+  const [recipient, setRecipient] = useState('')
+  const [hash, setHash] = useState('')
+
+  async function onConfirmation() {
+    try {
+      await unsubscribeRequest({
+        campaignId: +campaignId,
+        recipient,
+        hash,
+        version,
+      })
+      setUnsubscribed(true)
+    } catch (err) {
+      setErrorMsg(err.message)
+    }
+  }
+
+  async function isUrlValid({
+    campaignId,
+    recipient,
+    hash,
+    version,
+  }: {
+    campaignId: number
+    recipient: string
+    hash: string
+    version: string
+  }) {
+    try {
+      const isUnsub = await isUserUnsubscribed({
```

Comment:
why do we need to check this? the POST/PUT to unsubscribe should be idempotent

Reply:
It is mainly for UX, to check if the user is already unsubscribed. If he is unsubscribed, we won't show the confirmation button

Reply:
@lshaowei18 we've discussed during sync-up just now and we will remove the GET call. 

Even if the users have an invalid URL (invalid hash, etc) or have already unsubscribed previously, we will always show the unsub page with the unsub button first. The validation will only happen on POST to `/unsubscribe`. I'll make the backend changes in a follow up PR.